pkg/controller/pxc/status.go

```diff
@@ -360,5 +289,51 @@ func (r *ReconcilePerconaXtraDBCluster) appStatus(app api.StatefulApp, podSpec *
 		status.Status = api.AppStateReady
 	}
 
-	return status, nil
+	var condition api.ClusterCondition
+
+	status.Version = crStatus.Version
+	status.Image = crStatus.Image
+
+	switch status.Status {
+	case api.AppStateReady:
+		condition = api.ClusterCondition{
+			Status:             api.ConditionTrue,
+			Type:               api.ClusterReady,
+			Reason:             app.Name() + "Ready",
+			LastTransitionTime: metav1.NewTime(time.Now()),
+		}
+	case api.AppStateError:
+		condition = api.ClusterCondition{
+			Status:             api.ConditionTrue,
+			Message:            app.Name() + status.Message,
+			Reason:             app.Name() + "Error",
+			Type:               api.ClusterError,
+			LastTransitionTime: metav1.NewTime(time.Now()),
+		}
+	}
```

Comment:
It doesn't belong here.
1. You would check for status later after calling this method.
2. These conditions are being rewritten later in the giant switch.
3. Again you set condition here and later in the caller code - it should be done in one place.
4. This method is kinda agnostic to what app type it works with. But you are breaking it by dispatching the app type here.
5. Why it creates conditions for `Read` & `Error` and not for `Init`?

Do all this condition business in `updateStatus `.

Reply:
You're right, they're always overwritten in the caller code. Maybe we should remove this code altogether?